src/main/java/jenkins/plugins/openstack/compute/internal/Openstack.java

```diff
@@ -242,20 +314,67 @@ public int compare(Flavor o1, Flavor o2) {
         return keyPairs;
     }
 
-    public @CheckForNull String getImageIdFor(String name) {
-        Map<String, String> query = new HashMap<>(2);
-        query.put("name", name);
+    /**
+     * Finds the Id(s) of all active {@link Image}s with the given name or ID.
+     * If we have found multiple {@link Image}s then they will be listed in
+     * ascending date order (oldest first).
+     * 
+     * @param nameOrId
+     *            The name or ID of zero or more images.
+     * @return A non-null list. It may contain zero, one or multiple IDs.
+     */
+    public @Nonnull List<String> getImageIdsFor(String nameOrId) {
+        final Collection<Image> sortedObjects = new TreeSet<>(IMAGE_DATE_COMPARATOR);
+        final Map<String, String> query = new HashMap<>(2);
+        query.put("name", nameOrId);
         query.put("status", "active");
-
-        List<? extends Image> images = clientProvider.get().images().listAll(query);
-        if (images.size() > 0) {
-            // Pick one at random to point out failures ASAP
-            return images.get(new Random().nextInt(images.size())).getId();
+        final List<? extends Image> findByName = clientProvider.get().images().listAll(query);
+        sortedObjects.addAll(findByName);
+        if (nameOrId.matches("[0-9a-f-]{36}")) {
+            final Image findById = clientProvider.get().images().get(nameOrId);
+            if (findById != null && findById.getStatus() == Image.Status.ACTIVE) {
+                sortedObjects.add(findById);
+            }
         }
+        final List<String> ids = new ArrayList<>();
+        for (Image i : sortedObjects) {
+            ids.add(i.getId());
+        }
+        return ids;
+    }
 
-        if (name.matches("[0-1a-f-]{36}")) return name;
-
-        return null;
+    /**
+     * Finds the Id(s) of all active {@link VolumeSnapshot}s with the given name
+     * or ID. If we have found multiple {@link VolumeSnapshot}s then they will
+     * be listed in ascending date order (oldest first).
+     * 
+     * @param nameOrId
+     *            The name or ID of zero or more images.
+     * @return A non-null list. It may contain zero, one or multiple IDs.
+     */
+    public @Nonnull List<String> getVolumeSnapshotIdsFor(String nameOrId) {
+        final Collection<VolumeSnapshot> sortedObjects = new TreeSet<>(VOLUMESNAPSHOT_DATE_COMPARATOR);
+        // OpenStack block-storage/v3 API doesn't allow us to filter by name, so fetch all and search.
+        final Map<String, Collection<VolumeSnapshot>> allVolumeSnapshots = getVolumeSnapshots();
+        final Collection<VolumeSnapshot> findByName = allVolumeSnapshots.get(nameOrId);
+        if (findByName != null) {
+            for (VolumeSnapshot o : findByName) {
+                if (o.getStatus() == Volume.Status.AVAILABLE) {
```

Comment:
Can we move this filtering to `getVolumeSnapshots`? I do not think we do care for snapshots in any other state.